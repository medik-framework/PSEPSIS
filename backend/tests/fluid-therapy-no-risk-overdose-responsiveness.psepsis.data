{
  "fluidOverdoseRisks": null,
  "fluidOverdoseSigns": null,
  "Normal Saline": 20
}
