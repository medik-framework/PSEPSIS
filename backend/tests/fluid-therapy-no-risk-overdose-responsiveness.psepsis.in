[
  {
    "id": "tabletApp",
    "action": "broadcast",
    "eventName": "StartScreening",
    "eventArgs": []
  },
  {
    "id": "tabletApp",
    "action": "broadcast",
    "eventName": "AgeEntered",
    "eventArgs": [300],
    "waitFor":
     {
       "name": "Instruct",
       "args": ["get age"]
     }
  },
  {
    "id": "tabletApp",
    "action": "broadcast",
    "eventName": "StartFluidTherapy",
    "eventArgs": [],
    "waitFor":
     {
       "name": "Instruct",
       "args": ["get weight"]
     }
  },
  {
    "id": "tabletApp",
    "action": "broadcast",
    "eventName": "FluidOverloadRisksEntered",
    "eventArgs": [0],
    "waitFor" : {
      "name": "Instruct",
      "args": ["get fluid overload risks"]
    }
  },
  {
    "id": "tabletApp",
    "action": "broadcast",
    "eventName": "FluidRateEntered",
    "eventArgs": [300],
    "waitFor" : {
      "name": "Recommend"
    }
  },
  {
    "id": "tabletApp",
    "action": "broadcast",
    "eventName": "ConfirmNormalSalineAdministered",
    "eventArgs": [],
    "updateData": [
      {
        "name": "Normal Saline Count",
        "args": 1
      }, {
        "name": "Normal Saline",
        "args": 20
      }
    ],
    "waitFor" : {
      "name": "WaitFor"
    }
  },
  {
    "id": "tabletApp",
    "action": "broadcast",
    "eventName": "FluidResponsivenessEntered",
    "eventArgs": [false],
    "waitFor" : {
      "name": "Instruct",
      "args": ["get responsiveness to fluids"]
    }
  },
  {
    "id": "tabletApp",
    "action": "broadcast",
    "eventName": "FluidOverloadSignsEntered",
    "eventArgs": [0],
    "waitFor" : {
      "name": "Instruct",
      "args": ["get fluid overload signs"]
    }
  },
  {
    "id": "tabletApp",
    "action": "broadcast",
    "eventName": "ConfirmRepeatFluidBolus",
    "eventArgs": [],
    "waitFor" : {
      "name": "Recommend",
      "args": ["get fluid next step"
              , "Responsiveness to fluid therapy: false"
              , "Fluid overload: false"
              , "Recommendation: One more fluid bolus."]
    }
  },
  {
    "action": "exit",
    "waitFor": {
      "name": "AlternativeDrugs"
    }
  }
]
