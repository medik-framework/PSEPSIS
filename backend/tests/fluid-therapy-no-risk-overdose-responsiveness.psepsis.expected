[
  {
    "id": "tabletApp",
    "interface": "TabletApp",
    "name": "Instruct",
    "args": ["get fluid overload risks"]
  },
  {
    "id": "tabletApp",
    "interface": "TabletApp",
    "name": "ConfirmFluidOverloadRisksEntered",
    "args": []
  },
  {
    "id": "tabletApp",
    "interface": "TabletApp",
    "name": "RecommendDrug",
    "args": [
      "Normal Saline",
      "Fluid Therapy",
      "20",
      "administer 20 ml/kg Normal Saline"
    ]
  },
  {
    "id": "tabletApp",
    "interface": "TabletApp",
    "name": "Instruct",
    "args": ["get responsiveness to fluids"]
  },
  {
    "id": "tabletApp",
    "interface": "TabletApp",
    "name": "Instruct",
    "args": ["get fluid overload signs"]
  },
  {
    "id": "datastore",
    "interface": "Datastore",
    "name": "Obtain",
    "args": ["Normal Saline"]
  },
  {
    "id": "tabletApp",
    "interface": "TabletApp",
    "name": "Recommend",
    "args": [
      "get fluid next step",
      "Responsiveness to fluid therapy: false",
      "Fluid overload: false",
      "Recommendation: repeat fluid bolus"
    ]
  },
  {
    "id": "tabletApp",
    "interface": "TabletApp",
    "name": "RecommendDrug",
    "args": [
      "Normal Saline",
      "Fluid Therapy",
      "20",
      "administer 20 ml/kg Normal Saline"
    ]
  }
]
